lib/model/autocomplete.dart

```diff
@@ -241,7 +241,15 @@ class MentionAutocompleteView extends ChangeNotifier {
         store: store);
       if (result != 0) return result;
     }
-    return compareByDms(userA, userB, store: store);
+    final result = compareByDms(userA, userB, store: store);
+    if (result != 0) return result;
+
+    if (!userA.isBot && userB.isBot) {
+      return -1;
+    } else if (userA.isBot && !userB.isBot) {
+      return 1;
+    }
```

Comment:
nit: Could be a small helper function like `compareByRecency` and `compareByDms`.